Some dll (System.Net.Http) references twice in csproj

Our solution targets .NET 4.6.1 and at the top of paket.dependencies we have Framework: net461 with redirects: on.
We reference  .Net and .Net Core packages:
```
nuget Microsoft.AspNet.WebApi
nuget Microsoft.EntityFrameworkCore.SqlServer

```

After packet intall , System.Net.Http referenced twice in .csproj .

Compilation Error: 
```
error CS0433: The type StringContent' exists in both 'System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' and 'System.Net.Http, Version=4.1.1.1, Culture=neutral, 
PublicKeyToken=b03f5f7f11d50a3a
```

Bindings:
```
  <dependentAssembly>
    <Paket>True</Paket>
    <assemblyIdentity name="System.Net.Http" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
    <bindingRedirect oldVersion="0.0.0.0-65535.65535.65535.65535" newVersion="4.1.1.1" />
  </dependentAssembly>
```


Simple project is in attachement:
[PaketWebApp.zip](https://github.com/fsprojects/Paket/files/1015216/PaketWebApp.zip)

No workaround found for the moment.

--- a/src/Paket.Core/Installation/RestoreProcess.fs
+++ b/src/Paket.Core/Installation/RestoreProcess.fs
@@ -176,8 +176,9 @@ let CreateInstallModel(alternativeProjectRoot, root, groupName, sources, caches,
         let! (package, content) = ExtractPackage(alternativeProjectRoot, root, groupName, sources, caches, force, package, false)
         let model = 
                 InstallModel.CreateFromContent(
-                    package.Name, package.Version, 
-                    package.Settings.FrameworkRestrictions |> Requirements.getExplicitRestriction, 
+                    package.Name, 
+                    package.Version, 
+                    Requirements.getExplicitRestriction package.Settings.FrameworkRestrictions, 
                     content.Force())
         return (groupName,package.Name), (package,model)
     }

--- a/src/Paket.Core/PaketConfigFiles/ProjectFile.fs
+++ b/src/Paket.Core/PaketConfigFiles/ProjectFile.fs
@@ -785,8 +785,16 @@ module ProjectFile =
                             | "$(TargetFrameworkIdentifier) == 'true'" -> "true"
                             | _ -> condition
 
-                        let frameworkReferences = libFolder.FolderContents.FrameworkReferences |> Seq.sortBy (fun (r) -> r.Name) |> Seq.toList
-                        let libraries = libFolder.FolderContents.Libraries |> Seq.sortBy (fun (r) -> r.Path) |> Seq.toList
+                        let frameworkReferences = 
+                            libFolder.FolderContents.FrameworkReferences 
+                            |> Seq.sortBy (fun r -> r.Name) 
+                            |> Seq.toList
+
+                        let libraries =
+                            libFolder.FolderContents.Libraries 
+                            |> Seq.sortBy (fun r -> r.Path) 
+                            |> Seq.toList
+
                         let assemblyTargets = ref libFolder.Targets
                         let duplicates = HashSet<_>()
                         for frameworkAssembly in frameworkReferences do
@@ -799,7 +807,8 @@ module ProjectFile =
                             [condition,createItemGroup libFolder.Targets frameworkReferences libraries,false]
                         else
                             let specialFrameworkAssemblies, rest =
-                                frameworkReferences |> List.partition (fun fr -> duplicates.Contains fr.Name)
+                                frameworkReferences 
+                                |> List.partition (fun fr -> duplicates.Contains fr.Name)
 
                             match PlatformMatching.getCondition referenceCondition allTargets !assemblyTargets with
                             | "" -> [condition,createItemGroup libFolder.Targets rest libraries,false]
@@ -1095,10 +1104,21 @@ module ProjectFile =
 
         let usedFrameworkLibs = HashSet<TargetProfile*string>()
 
-        completeModel
-        |> Seq.filter (fun kv -> usedPackages.ContainsKey kv.Key)
-        |> Seq.sortBy (fun kv -> let group, packName = kv.Key in group.CompareString, packName.CompareString)
-        |> Seq.map (fun kv -> 
+        let httpPackage = PackageName "System.Net.Http"
+        let filteredModel =
+            completeModel
+            |> Map.filter (fun kv _ -> usedPackages.ContainsKey kv)
+
+        
+        filteredModel
+        |> Seq.sortBy (fun kv -> 
+                let group, packName = kv.Key
+                if packName = httpPackage then
+                    for t in KnownTargetProfiles.AllProfiles do
+                        usedFrameworkLibs.Add(t,"System.Net.Http") |> ignore
+
+                group.CompareString, packName.CompareString)
+        |> Seq.map (fun kv ->
             deleteCustomModelNodes (snd kv.Value) project
             let installSettings = snd usedPackages.[kv.Key]
             let restrictionList = 

--- a/src/Paket.Core/PaketConfigFiles/ProjectFile.fs
+++ b/src/Paket.Core/PaketConfigFiles/ProjectFile.fs
@@ -1104,11 +1104,13 @@ module ProjectFile =
 
         let usedFrameworkLibs = HashSet<TargetProfile*string>()
 
-        let httpPackage = PackageName "System.Net.Http"
+        let specialPackagesWithFrameworkConflictLibs = 
+            [PackageName "System.Net.Http"] // see https://github.com/fsprojects/Paket/issues/2352
+            |> Set.ofList
+
         let filteredModel =
             completeModel
             |> Map.filter (fun kv _ -> usedPackages.ContainsKey kv)
-
         
         filteredModel
         |> Seq.map (fun kv -> 
@@ -1124,11 +1126,13 @@ module ProjectFile =
                         .FilterExcludes(installSettings.Excludes)
                         .RemoveIfCompletelyEmpty()
 
-                let group, packName = kv.Key
-                if packName = httpPackage then
+                let _, packageName = kv.Key
+                if specialPackagesWithFrameworkConflictLibs.Contains packageName then
                     for t in KnownTargetProfiles.AllProfiles do
-                        if (projectModel.GetLibReferenceFiles t) |> Seq.exists (fun t -> t.Name = "System.Net.Http.dll") then
-                            usedFrameworkLibs.Add(t,"System.Net.Http") |> ignore
+                        if (projectModel.GetLibReferenceFiles t) 
+                           |> Seq.exists (fun t -> t.Name = packageName.ToString() + ".dll") 
+                        then
+                            usedFrameworkLibs.Add(t,packageName.ToString()) |> ignore
 
                 kv,installSettings,restrictionList,projectModel)
         |> Seq.sortBy (fun (kv,_,_,_) -> 

--- a/src/Paket.Core/PaketConfigFiles/ProjectFile.fs
+++ b/src/Paket.Core/PaketConfigFiles/ProjectFile.fs
@@ -1111,26 +1111,30 @@ module ProjectFile =
 
         
         filteredModel
-        |> Seq.sortBy (fun kv -> 
+        |> Seq.map (fun kv -> 
+                deleteCustomModelNodes (snd kv.Value) project
+                let installSettings = snd usedPackages.[kv.Key]
+                let restrictionList = 
+                    installSettings.FrameworkRestrictions 
+                    |> getExplicitRestriction
+
+                let projectModel =
+                    (snd kv.Value)
+                        .ApplyFrameworkRestrictions(restrictionList)
+                        .FilterExcludes(installSettings.Excludes)
+                        .RemoveIfCompletelyEmpty()
+
                 let group, packName = kv.Key
                 if packName = httpPackage then
                     for t in KnownTargetProfiles.AllProfiles do
-                        usedFrameworkLibs.Add(t,"System.Net.Http") |> ignore
+                        if (projectModel.GetLibReferenceFiles t) |> Seq.exists (fun t -> t.Name = "System.Net.Http.dll") then
+                            usedFrameworkLibs.Add(t,"System.Net.Http") |> ignore
 
+                kv,installSettings,restrictionList,projectModel)
+        |> Seq.sortBy (fun (kv,_,_,_) -> 
+                let group, packName = kv.Key
                 group.CompareString, packName.CompareString)
-        |> Seq.map (fun kv ->
-            deleteCustomModelNodes (snd kv.Value) project
-            let installSettings = snd usedPackages.[kv.Key]
-            let restrictionList = 
-                installSettings.FrameworkRestrictions 
-                |> getExplicitRestriction
-
-            let projectModel =
-                (snd kv.Value)
-                    .ApplyFrameworkRestrictions(restrictionList)
-                    .FilterExcludes(installSettings.Excludes)
-                    .RemoveIfCompletelyEmpty()
-
+        |> Seq.map (fun (kv,installSettings,restrictionList,projectModel) ->
             if directPackages.ContainsKey kv.Key then
                 let targetProfile = getTargetProfile project 
                 if isTargetMatchingRestrictions(restrictionList,targetProfile) then

--- a/src/Paket/Paket.fsproj
+++ b/src/Paket/Paket.fsproj
@@ -32,7 +32,7 @@
     <StartProgram>paket.exe</StartProgram>
     <StartAction>Project</StartAction>
     <StartArguments>update</StartArguments>
-    <StartWorkingDirectory>C:\temp\Library2</StartWorkingDirectory>
+    <StartWorkingDirectory>C:\temp\fixed</StartWorkingDirectory>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
     <Optimize>true</Optimize>

